File: app/src/main/java/add/project/micro/co/th/projectmicroprocessor/MyHolder.java
package add.project.micro.co.th.projectmicroprocessor;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;


public class MyHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
     ImageView image ;
    TextView statusView;
    ImageView arrowRight;
    private ItemClickListener itemClickListener;
    public MyHolder(View itemView) {
        super(itemView);
        image =  itemView.findViewById(R.id.im_machine);
        statusView =  itemView.findViewById(R.id.tv_status_first);
        arrowRight =  itemView.findViewById(R.id.im_ic_arrow_right);

        itemView.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {
        this.itemClickListener.onItemClick(v,getLayoutPosition());

    }
    public void setItemClickListener(ItemClickListener ic) {
        this.itemClickListener = ic;
    }
}

File: app/src/main/java/add/project/micro/co/th/projectmicroprocessor/recyclerview/MyHolder.kt
package add.project.micro.co.th.projectmicroprocessor.recyclerview

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import add.project.micro.co.th.projectmicroprocessor.R


class MyHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
    var image: ImageView = itemView.findViewById(R.id.im_machine)
    var statusView: TextView = itemView.findViewById(R.id.tv_status_first)
    var arrowRight: ImageView = itemView.findViewById(R.id.im_ic_arrow_right)
    private var itemClickListener: ItemClickListener? = null

    init {

        itemView.setOnClickListener(this)

    }

    override fun onClick(v: View) {
        this.itemClickListener!!.onItemClick(v, layoutPosition)

    }

    fun setItemClickListener(ic: ItemClickListener) {
        this.itemClickListener = ic
    }
}
